src/utils/public/apr.ts

```diff
@@ -0,0 +1,99 @@
+import { BN } from "@project-serum/anchor";
+import { PublicKey } from "@solana/web3.js";
+import Decimal from "decimal.js";
+import { ZERO_SLIPPAGE } from "../../constants/public";
+import { getRemoveLiquidityQuote } from "../../position/quotes/remove-liquidity";
+import { PoolData, PoolRewardInfo } from "../../types";
+import { TokenUSDPrices } from "../token-price";
+import { DecimalUtil } from "./decimal-utils";
+
+export type EstimatedAprs = {
+  fee: number;
+  rewards: number[];
+};
+
+export const ZERO_APR = {
+  fee: 0,
+  rewards: [0, 0, 0],
+};
+
+export function estimateAprsForPriceRange(
+  pool: PoolData,
+  // TODO: should this actually be fetched/shared?
+  // There's a weird in/out/in dependency here on prices/fees
```

Comment:
Hmm, so it looks like [getTokenUSDPrices](https://github.com/orca-so/whirlpool-sdk/blob/main/src/utils/token-price.ts#L23) caches the only async calls that it makes, so i think it's pretty reasonable to make the call in here as well.

Reply:
This only takes into account the whirlpools and not standard pools; for rewards where the token is not listed in whirlpools directly, the price cannot be derived from this call.